### DISCUSSION TITLE
Garth vs python garminconnect

### DISCUSSION THREAD

**Discussion by AnonJohn** [matin/garth#20] [General]
Hi Matin— Garth is now integrated into garminconnect. Yay! But this leads me to a question: What is your long-term vision for Garth? How do you see it as different from the older, excellent library? The answer to "why write another" on the home has two elements: modernized authentication and data analysis (e.g. colabs). The authentication is now included into garminconnect.

Do you plan to implement the other data endpoints available in garminconnect? (Only some are natively available in Garth). Keep building out cool examples of data analysis or frameworks for data analysis?

It feels like garth has accomplished everything it initially set out to do — congratulations, thank you! So I'm wondering where you see it going now ...

**Comment by matin** [comment]
Great to hear you're using Garth!

I'm working on enabling more methods of advanced data analysis. Colabs were the first step, but it's fairly time consuming to write them.

My most recent Colab allows you to [download consolidated stats](https://colab.research.google.com/github/matin/garth/blob/main/colabs/chatgpt_analysis_of_stats.ipynb). You can then feed the stats into ChatGPT with Advanced Data Analysis enabled.

Compare all my code in the [Colab for stress trends](https://colab.research.google.com/github/matin/garth/blob/main/colabs/stress.ipynb) vs asking ChatGPT to do the same thing:

<img width="492" alt="image" src="https://github.com/matin/garth/assets/98985/2bfd319e-5bc4-456e-9848-70ef0a854e31">

ChatGPT is writing the Python code and running in their Code Interpreter (essentially a hosted Jupyter Notebook like Google Colab).

Give it a try!

I'm still working on ways to make it easier to analyze the data with ChatGPT and providing prompt examples. The next step is to add detailed sleep data (in addition to just sleep quality) and other metrics like HRV.

Just as manually writing Colabs helped educate what I'm doing now, what I'm doing with ChatGPT's Advanced Data Analysis will inform what I do next.

Garth was meant as a personal project, but I'm happy Garth has been valuable to others. It's incredible how fast it's reached [50k+ downloads / month](https://www.pepy.tech/projects/garth) and growing.

To turn the question around, what would you like to see from Garth?

**Reply by tripmat** [reply]
Hi, thanks for your time and for garth. How can I access body composition/weight data and daily steps so that I can run correlations?

**Reply by matin** [reply]
> Hi, thanks for your time and for garth. How can I access body composition/weight data and daily steps so that I can run correlations?

@tripmat just saw this. you should create ticket for this. I'll add it
